terraform/deployments/ga4-analytics/service_account.tf

```
@@ -0,0 +1,25 @@
+resource "google_service_account" "sa--ga4-analytics-352613" {
+  account_id                   = "ga4-analytics-352613"
+  display_name                 = "App Engine default service account"
+  create_ignore_already_exists = "true"
+}
+
+resource "google_service_account" "sa--ga-database" {
+  account_id                   = "ga-database"
+  description                  = "Account for analytics_settings_database"
+  display_name                 = "ga-database"
+  create_ignore_already_exists = "true"
+}
+
+resource "google_service_account" "sa--ga4-user-admin" {
+  account_id                   = "ga4-user-admin"
+  display_name                 = "GA4 User Admin"
+  create_ignore_already_exists = "true"
+}
+
+resource "google_service_account" "sa--search-analytics-pipeline" {
```

Nit: 
```suggestion
resource "google_service_account" "search_analytics_pipeline" {
```

Don't think the "sa--" prefix adds anything to the resources names. Also use "_" rather than "-"s